### PR TITLE
add sorting + proper 'show more' button in ScreeningRoom

### PR DESCRIPTION
**Pull request by mike-lvov**
Adds sorting + proper show more button logic in screening room

### CODE REVIEW

**Review by yurii-lubynets — APPROVED** [review]

File: src/components/templates/ScreeningRoom/useScreeningRoom.ts [hunk 1]
```diff
@@ -119,24 +120,21 @@ export const useScreeningRoom = (screeningRoomVenueId: string) => {
...
-  }, [
-    searchQuery,
-    fuseVideos,
-    filteredVideosBySubCategory,
-    displayedVideosAmount,
-  ]);
+  }, [searchQuery, fuseVideos, filteredVideosBySubCategory]);
+
+  const displayedVideos = searchedVideos
+    .sort((a, b) => alphabeticalOrderSort(a.title, b.title))
```
Comment (sunny-viktoryia) [inline src/components/templates/ScreeningRoom/useScreeningRoom.ts:131]: can we use `a.title.localeCompare(b.title)` here?

see: https://stackoverflow.com/a/2167619

Comment (mike-lvov) [reply src/components/templates/ScreeningRoom/useScreeningRoom.ts]: @sunny-viktoryia good catch!

Fixed in https://github.com/sparkletown/sparkle/pull/1464/commits/b406c60efbda905b3a7f0fca95cb9633c4496fab

File: src/components/templates/ScreeningRoom/useScreeningRoom.ts [hunk 2]
```diff
@@ -70,7 +70,7 @@ export const useScreeningRoom = (screeningRoomVenueId: string) => {
   const increaseDisplayedVideosAmount = useCallback(() => {
     setDisplayedVideosAmount(
       (prevPostersNumber) =>
-        prevPostersNumber + DEFAULT_DISPLAYED_POSTER_PREVIEW_COUNT
+        prevPostersNumber + DEFAULT_DISPLAYED_VIDEO_PREVIEW_COUNT
     );
```
Comment (0xdevalias) [inline src/components/templates/ScreeningRoom/useScreeningRoom.ts:74]: [nit] change `prevPostersNumber` to `prevVideosNumber` or similar.

**Review by 0xdevalias — APPROVED** [review]
One tiny nit change, then LGTM!

It might also be good to add some `// @debt` comments pointing out that this code seems to be almost a direct duplication from the `PosterHall`, and to handle cleaning that up/de-duplicating it in future.

File: src/settings.ts [hunk 3]
```diff
@@ -585,6 +585,7 @@ export const POSTERPAGE_MAX_VIDEO_PARTICIPANTS = 10;
 export const SEARCH_DEBOUNCE_TIME = 200; // ms
 
 export const DEFAULT_DISPLAYED_POSTER_PREVIEW_COUNT = 48;
```
Comment (denisdimitrov) [inline src/settings.ts:587]: Please remove the old `const` if it's not being used anymore.

File: src/components/templates/ScreeningRoom/useScreeningRoom.ts [hunk 4]
```diff
@@ -119,24 +119,21 @@ export const useScreeningRoom = (screeningRoomVenueId: string) => {
...
-    displayedVideosAmount,
-  ]);
+  }, [searchQuery, fuseVideos, filteredVideosBySubCategory]);
+
+  const displayedVideos = searchedVideos
+    .sort((a, b) => a.title.localeCompare(b.title))
+    .slice(0, displayedVideosAmount);
 
-  const hasHiddenVideos = searchedVideos.length > displayedVideosAmount;
+  const hasHiddenVideos = searchedVideos.length > displayedVideos.length;
```
Comment (sunny-viktoryia) [inline src/components/templates/ScreeningRoom/useScreeningRoom.ts:133]: and can you explain why was that changed?
I guess `displayedVideosAmount` should have the same value. 🤔

File: src/settings.ts (hunk 3 above)
Comment (0xdevalias) [reply src/settings.ts:587]: I expect this is still being used in the `PosterHall`

File: src/components/templates/ScreeningRoom/useScreeningRoom.ts (hunk 4 above)
Comment (0xdevalias) [reply src/components/templates/ScreeningRoom/useScreeningRoom.ts:133]: Since `displayedVideos` created from a `.slice(0, displayedVideosAmount)` it should at most have an array length of `displayedVideosAmount`; but I expect it could possibly have less elements than that if `displayedVideosAmount` is set to a value higher than the total length of `searchedVideos`

Comment (mike-lvov) [reply src/components/templates/ScreeningRoom/useScreeningRoom.ts:133]: @sunny-viktoryia that logic was broken. Now it's fixed

File: src/settings.ts (hunk 3 above)
Comment (mike-lvov) [reply src/settings.ts:587]: @denisdimitrov that const is used in PosterHall
